File: codility/4.CountingElements/PermCheck/PermCheck.java
/**
 * Author: Liang Lin
 * Email: lianglin@outlook.com
 * Date: 2016-03-26
 */
import java.util.BitSet;
Class Solution {
    public int solution(int[] A) {
        int n = A.length;
        BitSet B = new BitSet(n);
        for(int i=0; i<n; i++) {
            if(A[i]>n || B.get(A[i]-1)) {
                return 0;
            } else {
                B.set(A[i]-1);
            }
        }
        return 1;
    }
}

File: codility/4.CountingElements/PermCheck/PermCheck.py
# Author: Liang Lin
# Email: lianglin@outlook.com
# Date: 2016-03-26
class Solution:
  def solution(self, A):
    n = len(A)
    B = [False] * n
    for i in range(n):
        if A[i] > n or A[i] < 1 or B[A[i]-1]:
            return 0
        else:
            B[A[i]-1] = True
    return 1
